src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -286,45 +291,769 @@ public List<Element> getMembers(Kind kind) {
+        var t = new OverrideSequence(fixedSequence, 0);
+        while (true) {
+            overriddenMethods.put(t.getMethod(), t);
+            if (!t.hasLessSpecific()) {
+                break;
+            } else {
+                t = t.lessSpecific();
+            }
+        }
+    }
+
+    private Map<TypeElement, Integer> createSupertypeOrderMap(TypeElement e) {
+        // replicate legacy order
+        var m = new HashMap<TypeElement, Integer>();
+        if (e.getKind().isClass()) {
+            // ascend by superclasses
+            var c = e;
+            while (true) {
+                m.put(c, m.size());
+                if (c.getSuperclass().getKind() == TypeKind.NONE)
+                    break;
+                c = (TypeElement) ((DeclaredType) c.getSuperclass()).asElement();
+            }
+        }
+        if (e.getKind().isInterface()) {
+            m.put(e, m.size());
+        }
+        var superinterfaces = getAllInterfaces(e);
+        for (var s : superinterfaces) {
+            TypeElement e1 = (TypeElement) ((DeclaredType) s).asElement();
+            assert !m.containsKey(e1);
+            m.put(e1, m.size());
+        }
+        // patch for interfaces
+        if (e.getKind().isInterface()) {
+            // java.lang.Object is the last supertype in a graph of supertypes
+            // for an interface
+            var prev = m.put(elements().getTypeElement("java.lang.Object"), m.size());
+            assert prev == null;
+        }
+        return m;
+    }
+
+    public Set<TypeMirror> getAllInterfaces(TypeElement te) {
+        Set<TypeMirror> results = new LinkedHashSet<>();
+        addSuperInterfaces(te.asType(), results, new HashSet<>());
+        return results;
+    }
+
+    private void addSuperInterfaces(TypeMirror type, Set<TypeMirror> results, Set<Element> visited) {
+        TypeMirror superType = null;
+        for (TypeMirror t : types().directSupertypes(type)) {
+            if (types().isSameType(t, utils.getObjectType()))
+                continue;
+            TypeElement e = asTypeElement(t);
+            if (utils.isPlainInterface(e)) {
+                if (!visited.add(e)) {
+                    continue; // seen it before
+                }
+                results.add(t);
+                addSuperInterfaces(t, results, visited);
+            } else {
+                // there can be at most one superclass and it is not null
+                assert superType == null && t != null : superType;
+                // Save the supertype for later.
+                superType = t;
+            }
+        }
+        // Collect the super-interfaces of the supertype.
+        if (superType != null)
+            addSuperInterfaces(superType, results, visited);
+    }
+
+
+
+    private com.sun.tools.javac.util.List<OverrideData> findOverriddenBy(ExecutableElement m) {
+        return findOverriddenBy(m, (DeclaredType) te.asType(), com.sun.tools.javac.util.List.nil(), new HashSet<>());
```

Comment:
What is pushing you into having to use `javac`'s internal `List` class?

Reply:
IIRC, it's a combination of vestigial reasons and the fact that there's the `reverse()` method, whose cousin SequencedCollection.reversed() is sadly unavailable in langtools in JDK 21 because of "JDK N-1".

I'll try to get rid of javac's list here.